### PR TITLE
chore(python): Move References from `Writer` to `AstNode`

### PR DESCRIPTION
**Pull request by noanflaherty**
In trying to integrate fern's codegen with vellum, I've found it problematic for the `Writer` to be the thing that keeps track of references and instead, for it to be much more natural for an `AstNode` to keep track of its own references.

This came to a head with generating files and handling imports. `PythonFile` relied on `Writer`'s references to generate imports, but this depended on all of the file's statements being written first. This means only one of two bad options were possible:
1. Write imports then statements -> This led to empty imports becuase the writer didn't have any references yet because the statements hadn't been written yet; or
2. Write statements then imports –> This led to imports being underneath statements

With this fix, it both feels more natural and makes file writing possible.

### CODE REVIEW

File: generators/pythonv2/codegen/src/ast/Method.ts [hunk 1]
```diff
@@ -28,8 +28,6 @@ export declare namespace Method {
         docstring?: string;
         /* The type of the method if defined within the context of a class */
         type?: ClassMethodType;
-        /* The class this method belongs to, if any */
-        classReference?: Reference;
```
Comment (noanflaherty) [inline generators/pythonv2/codegen/src/ast/Method.ts:32]: Semi-unrelated change. Method's weren't using this yet.

File: generators/pythonv2/codegen/src/ast/__test__/__snapshots__/PythonFile.test.ts.snap [hunk 2]
```diff
@@ -6,6 +6,15 @@ exports[`PythonFile > Add a Method 1`] = `
 "
 `;
 
+exports[`PythonFile > Add a class that inherits from a class imported from another file 1`] = `
+"from ..test_module.base import BaseClass
+
+
+class DerivedClass(BaseClass):
+    pass
```
Comment (noanflaherty) [inline generators/pythonv2/codegen/src/ast/__test__/__snapshots__/PythonFile.test.ts.snap:14]: A test that proves this works now.

**Review by dsinghvi — APPROVED** [review]
